app/js/lib/speech/synthesis.js

```diff
@@ -0,0 +1,74 @@
+'use strict';
+
+const p = Object.freeze({
+  // Properties
+  synthesis: Symbol('synthesis'),
+  supportsSynthesis: Symbol('supportsSynthesis'),
+  voice: Symbol('voice'),
+
+  // Methods
+  populateVoice: Symbol('populateVoice'),
+});
+
+export default class SpeechSynthesis {
+  constructor() {
+    const synthesis = window.speechSynthesis;
+
+    this[p.supportsSynthesis] = !!synthesis;
+
+    if (this[p.supportsSynthesis]) {
+      this[p.synthesis] = synthesis;
+      this[p.synthesis].onvoiceschanged = this[p.populateVoice].bind(this);
```

Comment:
It doesn't work in Chrome for me, we have to wait for the callback to be executed. I'll change that to call `populateVoice` here too and make sure it doesn't get executed twice.